pkg/agent/controller/networkpolicy/networkpolicy_controller_test.go

```diff
@@ -436,3 +456,102 @@ func TestAddNetworkPolicyWithMultipleRules(t *testing.T) {
+		t.Fatalf("Expected no update, got %v", ruleID)
+	case <-time.After(time.Millisecond * 100):
+	}
+	checkNetworkPolicyMetrics(t, controller)
+
+	// Test deleting policy1
+	networkPolicyWatcher.Delete(newNetworkPolicy("policy1", []string{}, []string{}, []string{}, nil))
+	select {
+	case ruleID := <-reconciler.updated:
+		t.Fatalf("Expected no update, got %v", ruleID)
+	case <-time.After(time.Millisecond * 100):
+	}
+	checkNetworkPolicyMetrics(t, controller)
+
+	// Test deleting policy2
+	networkPolicyWatcher.Delete(newNetworkPolicy("policy2", []string{}, []string{}, []string{}, nil))
+	select {
+	case ruleID := <-reconciler.updated:
+		t.Fatalf("Expected no update, got %v", ruleID)
+	case <-time.After(time.Millisecond * 100):
+	}
+	checkNetworkPolicyMetrics(t, controller)
+}
+
+func checkNetworkPolicyMetrics(t *testing.T, controller *Controller) {
+	expectedEgressNetworkPolicyRuleCount := `
+    # HELP antrea_agent_egress_networkpolicy_rule_count [STABLE] Number of egress networkpolicy rules on local node which are managed by the Antrea Agent.
+    # TYPE antrea_agent_egress_networkpolicy_rule_count gauge
+	`
+
+	expectedIngressNetworkPolicyRuleCount := `
+    # HELP antrea_agent_ingress_networkpolicy_rule_count [STABLE] Number of ingress networkpolicy rules on local node which are managed by the Antrea Agent.
+    # TYPE antrea_agent_ingress_networkpolicy_rule_count gauge
+	`
+
+	expectedNetworkPolicyCount := `
+    # HELP antrea_agent_networkpolicy_count [STABLE] Number of networkpolicies on local node which are managed by the Antrea Agent.
+    # TYPE antrea_agent_networkpolicy_count gauge
+	`
+
+	ingressRuleCount := 0
+	egressRuleCount := 0
+
+	networkpolicies := controller.GetNetworkPolicies("")
+	for _, networkpolicy := range networkpolicies {
+		for _, rule := range networkpolicy.Rules {
+			if rule.Direction == v1beta1.DirectionIn {
+				ingressRuleCount++
+			} else {
+				egressRuleCount++
+			}
+		}
+	}
+
+	expectedEgressNetworkPolicyRuleCount = expectedEgressNetworkPolicyRuleCount + fmt.Sprintf("antrea_agent_egress_networkpolicy_rule_count %d\n", egressRuleCount)
+	expectedIngressNetworkPolicyRuleCount = expectedIngressNetworkPolicyRuleCount + fmt.Sprintf("antrea_agent_ingress_networkpolicy_rule_count %d\n", ingressRuleCount)
+	expectedNetworkPolicyCount = expectedNetworkPolicyCount + fmt.Sprintf("antrea_agent_networkpolicy_count %d\n", controller.GetNetworkPolicyNum())
+
+	assert.Equal(t, nil, testutil.GatherAndCompare(legacyregistry.DefaultGatherer, strings.NewReader(expectedEgressNetworkPolicyRuleCount), "antrea_agent_egress_networkpolicy_rule_count"))
+	assert.Equal(t, nil, testutil.GatherAndCompare(legacyregistry.DefaultGatherer, strings.NewReader(expectedIngressNetworkPolicyRuleCount), "antrea_agent_ingress_networkpolicy_rule_count"))
+	assert.Equal(t, nil, testutil.GatherAndCompare(legacyregistry.DefaultGatherer, strings.NewReader(expectedNetworkPolicyCount), "antrea_agent_networkpolicy_count"))
```

Comment:
please use `assert.NoError` for readability

Reply:
Thank you for your suggestion, I'll change.